### PR TITLE
Fix use of percentage-based input in manual grading

### LINKED ISSUE #7791: Manual grading issues for questions without points
As reported on Slack, the manual grading page is unable to submit a new grade if the question has no points. This is likely related to the handling of percentage, with division of zero involved in that case.

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Based on issue mentioned on Slack. Resolves two issues:

* Recent rubrics update broke the way manual input of points (particularly percentage) is handled, since the function relies on `this`, for which the function is no longer bound to. For safety and clarity this was replaced with an explicit argument.
* Resolves #7791, i.e., properly handles manual grading cases when the question is worth zero in total (i.e., no manual or auto points).

### CODE REVIEW

**Comment by jonatanschroeder** [comment]
Converting to draft to allow this to also handle #7791.

**Comment by jonatanschroeder** [comment]
This is now ready for review. The change also gets rid of percentage-based grading for questions with no points, since in this case percentage makes no sense. The use of local storage to handle percentage or points is ignored in this kind of question, since the processing is done in an event handler for the selector, and the selector does not exist.

File: apps/prairielearn/src/pages/instructorAssessmentManualGrading/instanceQuestion/gradingPointsSection.ejs [hunk 1]
```diff
@@ -3,21 +3,25 @@
...
           <span class="js-<%= type %>-score-value-info">
-            <span class="js-value-<%= type %>-points"><%= Math.round(points * 100) / 100 %></span> / <%= max_points %>
+            <span class="js-value-<%= type %>-points"></span> / <%= max_points %>
           </span>
         </span>
+        <% if (assessment_question.max_points) { %>
         <span class="js-manual-grading-percentage">
           <span class="js-<%= type %>-score-value-info">
-            <span class="js-value-<%= type %>-percentage"><%= Math.round(points * 10000 / (max_points || assessment_question.max_points)) / 100 %></span>%
+            <span class="js-value-<%= type %>-percentage"></span>%
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/pages/instructorAssessmentManualGrading/instanceQuestion/gradingPointsSection.ejs:21]: The initialization of these values is unnecessary, since JS already updates them when the page is loaded. Keeping it blank for simplicity and to avoid inconsistencies between the formatting done here and in JS.

File: apps/prairielearn/assets/scripts/instructorAssessmentManualGradingInstanceQuestion.js [hunk 2]
```diff
@@ -414,7 +414,7 @@ function computePointsFromRubric() {
       manualInput.value = manualPoints;
     }
   });
-  updatePointsView();
+  updatePointsView(this);
```
Comment (mwest1066) [inline apps/prairielearn/assets/scripts/instructorAssessmentManualGradingInstanceQuestion.js:417]: I personally find the use of `this` to be hard to follow in code that isn't a member function of a `class`. I like the fact that arrow functions have dropped this-binding, making it harder to write code that uses this-binding via `call`/`apply`/`bind`.

Could we instead pass the `item`/`input` (hopefully with a better name) as an argument to `computePointsFromRubric()` (using an arrow function in the `addEventListener()` calls that set up the handlers)? Then we can explicitly pass this `item` down to `updatePointsView()`.

File: apps/prairielearn/src/pages/instructorAssessmentManualGrading/instanceQuestion/gradingPointsSection.ejs (hunk 1 above)
Comment (mwest1066) [reply apps/prairielearn/src/pages/instructorAssessmentManualGrading/instanceQuestion/gradingPointsSection.ejs:21]: Sounds good.

Unrelated to the current PR: I find this pattern of dynamically constructing class names (or attribute names, etc) to make the code harder to follow because we can't grep for names. I generally prefer to either have all the code written out multiple times, once for each possible class name, or to store the entire class name as a variable with the full class names in the code:
```
if (type == 'auto') {
    valuePercentageClass = 'js-value-auto-percentage';
} else {
    valuePercentageClass = 'js-value-manual-percentage';
}
```

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/assets/scripts/instructorAssessmentManualGradingInstanceQuestion.js (hunk 2 above)
Comment (jonatanschroeder) [reply apps/prairielearn/assets/scripts/instructorAssessmentManualGradingInstanceQuestion.js]: Done in d9df5cbf2.

File: apps/prairielearn/src/pages/instructorAssessmentManualGrading/instanceQuestion/gradingPointsSection.ejs (hunk 1 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/pages/instructorAssessmentManualGrading/instanceQuestion/gradingPointsSection.ejs:21]: > Unrelated to the current PR: I find this pattern of dynamically constructing class names (or attribute names, etc) to make the code harder to follow because we can't grep for names. I generally prefer to either have all the code written out multiple times, once for each possible class name, or to store the entire class name as a variable with the full class names in the code:

Noted. Since this is not directly related to this PR I'll keep this as is and keep it in mind for future dev.
